source/event.hpp

```diff
@@ -34,7 +40,7 @@ struct Generate : public Event {
  */
 struct Arrive : public Event {
     // TODO: move implementation to .cpp or use existing if present
-    Arrive(ILink *a_link, Packet *a_packet) : link(a_link), packet(a_packet){};
+    Arrive(std::uint32_t a_time, ILink *a_link, Packet *a_packet) : link(a_link), packet(a_packet), Event(a_time) {};
```

Comment:
Let's use smart pointers:
```suggestion
    Arrive(std::uint32_t a_time, std::shared_ptr<ILink> a_link, std::shared_ptr<Packet> a_packet) : link(a_link), packet(a_packet), Event(a_time) {};
```

Reply:
What about weak_ptr instead of shared?
I dont think that Event owns Link or Device, it should use it if one exists

Reply:
Discussed on meeting. Summary: smart pointers will be implemented in all events later